src/test/java/com/google/devtools/build/lib/rules/objc/ObjcLibraryTest.java

```diff
@@ -1427,6 +1427,41 @@ public void testLegacyObjcProtoLibraryDoesNotCrash() throws Exception {
   }
 
   @Test
+  public void testObjcNamespacedImports() throws Exception {
+    useConfiguration(
+        "--crosstool_top=" + MockObjcSupport.DEFAULT_OSX_CROSSTOOL,
+        "--experimental_objc_crosstool=all",
+        "--experimental_enable_implicit_headermaps=true");
+    scratch.file(
+        "x/importer.m",
+        // By default, we would be able to import this header
+        // under the namespace of dep_liba.
+        "#import \"dep_liba/header.h\"",
+        // This header has a custom namespace
+        "#import \"some/headerb.h\"");
+    scratch.file(
+        "x/headerb.h");
+    scratch.file(
+        "x/BUILD",
+        "objc_library(",
+        "   name = 'objc',",
+        "   srcs = ['importer.m'],",
+        "   deps = [':dep_liba', ':dep_libb'],",
+        ")",
+        "objc_library(",
+        "   name = 'dep_liba',",
+        "   hdrs = ['header.h'],",
+        ")",
+        "objc_library(",
+        "   name = 'dep_libb',",
+        "   header_namespace = 'some',",
+        "   hdrs = ['headerb.h'],",
+        ")");
+
+    assertThat(getConfiguredTarget("//x:objc")).isNotNull();
```

Comment:
I didn't see a way to actually build `//x:objc` here, but that could be good to have it running end to end!

Reply:
The Java tests are not integration tests. You'd need to write an integration test to do an actual build.

Reply:
@ulfjack thanks for the feedback. I'm not too familiar with the integration testing system yet but will look into this.

Reply:
Well the easiest would be to take inspiration from our existing integration tests, e.g. https://github.com/bazelbuild/bazel/blob/master/src/test/shell/integration/cpp_test.sh